### PR TITLE
Cluster Debug

### PR DESCRIPTION
**Pull request by HURUIZHE**
Signed-off-by: HURUIZHE <553834398@qq.com>

### CODE REVIEW

**Comment by HURUIZHE** [comment]
why

**Comment by zzjin** [comment]
Check the linter error at:
```
  Running [/home/runner/golangci-lint-1.49.0-linux-amd64/golangci-lint run --out-format=github-actions --path-prefix=./controllers/infra --out-${NO_FUTURE}format colored-line-number] in [/home/runner/work/sealos/sealos/controllers/infra] ...
  controllers/infra/drivers/reconcile_test.go:4: File is not `goimports`-ed (goimports)
  	"github.com/labring/sealos/pkg/types/v1beta1"
```

File: controllers/cluster/applier/applier.go [hunk 1]
```diff
@@ -77,36 +77,38 @@ func (a *Applier) ReconcileCluster(infra *v1.Infra, hosts []v1.Hosts, cluster *c
...
-	if err = scpCluster(ssh, sealos, infra.Spec.SSH, "master"); err != nil {
+	if err = scpCluster(ssh, sealos, cluster.Spec.SSH, "master"); err != nil {
 		return fmt.Errorf("scp cluster master error :%v", err)
 	}
 	ipMasters, ipNodes := getIPs(master), getIPs(node)
 
-	fmt.Println("sudo sh masterDepl.sh master0 " + passwd + " " + ipMasters + " " + ipNodes)
-	//todo Add images
-	err = ssh.CmdAsync(sealos, "sudo sh masterDepl.sh master0 "+passwd+" "+ipMasters+" "+ipNodes, "")
+	fmt.Println("sudo sh masterDepl.sh master0 " + passwd + " " + ipMasters + " " + ipNodes + " " + images)
```
Comment (fanux) [inline controllers/cluster/applier/applier.go:96]: fmt.Printf()

File: controllers/cluster/applier/applier.go [hunk 2]
```diff
@@ -77,36 +77,38 @@ func (a *Applier) ReconcileCluster(infra *v1.Infra, hosts []v1.Hosts, cluster *c
...
 	}
-
 	return nil
 }
 
 func getImages(cluster *cv1.Cluster) string {
 	images := ""
-	for _, image := range cluster.Spec.Images {
-		images = images + image + " "
+	for i, image := range cluster.Spec.Images {
```
Comment (fanux) [inline controllers/cluster/applier/applier.go:106]: I don't quite understand why need this

File: controllers/cluster/applier/applier.go [hunk 3]
```diff
@@ -124,8 +126,8 @@ func getIPs(ips []string) string {
 }
 func makeClusterSingle(ssh ssh.Interface, ip string, passwd string, role string, images string) error {
 	fileName := role + "Depl.sh"
-	fmt.Println("sudo sh " + fileName + " master " + passwd)
-	err := ssh.CmdAsync(ip, "sudo sh "+fileName+" master "+passwd, "")
+	fmt.Println("sudo sh " + fileName + " master " + passwd + " " + images)
+	err := ssh.CmdAsync(ip, "sudo sh "+fileName+" master "+passwd+" "+images, "")
```
Comment (fanux) [inline controllers/cluster/applier/applier.go:130]: fmt.Sprintf("sudo sh %s master %s")...

File: controllers/cluster/applier/masterDepl.sh [hunk 4]
```diff
@@ -3,12 +3,20 @@ role=$1
 password=$2
 ipmasters=$3
 ipnodes=$4
+string=$5
+array=(${string//,/ })
```
Comment (fanux) [inline controllers/cluster/applier/masterDepl.sh:7]: ?

File: controllers/infra/drivers/aws/price.go [hunk 5]
```diff
@@ -8,5 +8,5 @@ type Price struct {
 // QueryPrice query infra price/hour
 func (pReq Price) QueryPrice(infra *v1.Infra) (int64, error) {
 	//TODO implement me
-	panic("implement me")
+	return 1, nil
```
Comment (fanux) [inline controllers/infra/drivers/aws/price.go:11]: func (infra *v1.Infra) QueryPrice() (int64, error) is better?

File: controllers/cluster/masterDepl.sh [hunk 6]
```diff
@@ -3,12 +3,20 @@ role=$1
 password=$2
 ipmasters=$3
 ipnodes=$4
```
Comment (cuisongliu) [inline controllers/cluster/masterDepl.sh:5]: using go template is better ?

Comment (HURUIZHE) [reply controllers/cluster/masterDepl.sh:5]: todo: use name instead of number

File: controllers/infra/drivers/aws/price.go (hunk 5 above)
Comment (HURUIZHE) [reply controllers/infra/drivers/aws/price.go:11]: ok

File: controllers/cluster/applier/masterDepl.sh (hunk 4 above)
Comment (HURUIZHE) [reply controllers/cluster/applier/masterDepl.sh:7]: images parameter. Modified

File: controllers/cluster/applier/applier.go (hunk 2 above)
Comment (HURUIZHE) [reply controllers/cluster/applier/applier.go:106]: array -> string(as parameter; split:",") -> array(in shell) -> string(split: " ")

Comment (HURUIZHE) [reply controllers/cluster/applier/applier.go:106]: here: array->string, to be parameter

File: controllers/cluster/applier/applier.go (hunk 1 above)
Comment (HURUIZHE) [reply controllers/cluster/applier/applier.go:96]: ok

File: controllers/infra/drivers/aws/price.go (hunk 5 above)
Comment (HURUIZHE) [reply controllers/infra/drivers/aws/price.go]: this function completed

**Review by fanux — APPROVED** [review]

**Review by fanux — APPROVED** [review]

**Pull request merged**